espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java

```diff
@@ -82,4 +85,31 @@ public Response serve(String uri, Method method, Map<String,
                     new AppiumResponse<>(AppiumStatus.UNKNOWN_ERROR, Log.getStackTraceString(e)));
         }
     }
+
+    public void start() throws IOException {
+        if(super.isAlive()) {
+            //kill the server if its already running
+            try {
+                super.stop();
+            } catch (Exception e) {
+                //ignore the exception
+            }
+        }
+        super.start(NanoHTTPD.SOCKET_READ_TIMEOUT, false);
+        logger.info(String.format("\nRunning Appium Espresso Server at port %d \n", DEFAULT_PORT));
+        router = new Router();
+    }
+
+    public void stop() {
+        super.stop();
+        logger.info(String.format("\nStopping Appium Espresso stop at port %d \n", DEFAULT_PORT));
+    }
+
+    public void makeRequestForServerToStop() {
+        isStopRequestReceived = true;
```

Comment:
should it be ever reset to false?

Reply:
The default value would be `false`, the moment `isStopRequestReceived ` updated which happens only when `DELETE` session requested, then immediatly [server](https://github.com/sravanmedarapu/appium-espresso-driver/blob/master/espresso-server/app/src/androidTest/java/io/appium/espressoserver/EspressoServerRunnerTest.java#L46)  will show down, so would never wanted to reset the `isStopRequestReceived` value.

Reply:
Sorry, not really sure I understand your question :(

Reply:
Hmm, it seems we have different understanings here. let me brief you what my understanding is 😊

The espresso server runs on the device and doesn't support parallel execution(i.e only one server instance will be running) in a single device. when `DELETE` session API invoked the `makeRequestForServerToStop()` gets executed and as an effect [server](https://github.com/sravanmedarapu/appium-espresso-driver/blob/master/espresso-server/app/src/androidTest/java/io/appium/espressoserver/EspressoServerRunnerTest.java#L49) shutdown and thats the end of application process life cycle. When a new session requested again a new process will be created and `isStopRequestReceived` will be initialized with default value i.e `false`.

Reply:
ok I got the point

Reply:
right, thanks for the clarification!